modules/assim.sequential/R/hop_test.R

```diff
@@ -0,0 +1,98 @@
+##' @title hop_test
+##' @name  hop_test
+##' @author Ann Raiho \email{araiho@@nd.edu}
+##' 
+##' @param settings    SDA PEcAn settings object
+##'
+##' @description Hop test. This script tests that the model successfully reads it's own restart and can restart without loss of information.
+##' 
+##' @return NONE
+##' @export
+##' 
+hop_test <- function(settings, ens.runid = NULL){
+  ##### 
+  ##### Variable to check
+  ##### 
+  hop_var <- lapply(settings$state.data.assimilation$state.variables,'[[','variable.name')
+  
+  ##### 
+  ##### Regular Run
+  ##### 
+  if(is.null(ens.runid)){
+    run.write.configs(settings, write = settings$database$bety$write)
+    
+    PEcAn.remote::start.model.runs(settings, settings$database$bety$write)
+    
+    ens.runid <- read.table(file.path(settings$rundir,'runs.txt'))
+  }
+  ens <- read.output(runid = ens.runid, 
+                     outdir = file.path(settings$outdir,'out', ens.runid), 
+                     start.year = lubridate::year(settings$run$start.date), 
+                     end.year = lubridate::year(settings$run$end.date), 
+                     variables = hop_var)
+  ##### 
+  ##### Create Empty Data
+  ##### 
+  reg_run_end <- lubridate::year(settings$run$end.date)
+  settings$run$end.date <- paste0(reg_run_end - 9,'/12/31') #changing the run end.date so that the .nc files get written after write_restart
+  
+  obs.mean <- obs.cov <- rep(list(NA),10)
+  names(obs.mean) <- names(obs.cov) <- paste0(lubridate::year(settings$run$end.date):reg_run_end,'/12/31')
+  
+  ##### 
+  ##### Run in SDA code with no data -- HOP Run
+  ##### 
+  PEcAn.assim.sequential::sda.enkf(settings = settings, obs.mean = obs.mean, obs.cov = obs.cov)
+  
+  hop.runid <- read.table(file.path(settings$rundir,'runs.txt'))
+  hop.ens <- read.output(runid = hop.runid, 
+                     outdir = file.path(settings$outdir,'out', hop.runid), 
+                     start.year = lubridate::year(settings$run$start.date), 
+                     end.year = reg_run_end, 
+                     variables = hop_var)
+  
+  save(list = ls(envir = environment(), all.names = TRUE), 
+       file = file.path(settings$outdir, "hop_test_output.Rdata"), envir = environment())
+  
+  ##### 
+  ##### Comparison: Hop Run versus Regular Run
+  ##### 
+  
+  plot_years <- lubridate::year(settings$run$start.date):reg_run_end
+  
+  pdf('hop_test_results.pdf')
+  par(mfrow=c(2,1))
+  for(p in seq_along(hop_var)){
+    
+    hop_var_use <- unlist(hop_var[p])
+    ens.plot <- ens[[hop_var_use]][(length(plot_years)-10):length(plot_years)]
+    hop.ens.plot <- hop.ens[[hop_var_use]][(length(plot_years)-10):length(plot_years)]
+    
+    plot(plot_years,
+         ens[[hop_var_use]],
```

Comment:
will x-y dims always match? i.e. can't outputs be sub-annual?

Reply:
I didn't set this up to be sub-annual. Do you have any ideas for how to do that?

Reply:
I think read output returns a time column and a year column, maybe utilize that?

Reply:
I think it's fine for this version to be annual and we can generalize as we apply the hop test to other models